runtime/src/iree/tooling/context_util.c

```diff
@@ -169,6 +169,42 @@ iree_status_t iree_tooling_load_modules_from_flags(
   return iree_ok_status();
 }
 
+//===----------------------------------------------------------------------===//
+// HAL module device selection policy
+//===----------------------------------------------------------------------===//
+
+IREE_FLAG(int32_t, device_lead_allocator, 0,
+          "Device ordinal of the lead device that will be used for allocations "
+          "when more than one device is available. Only functions when there "
+          "are selection requests including all devices and otherwise the "
+          "first device in the list will be selected.");
```

Comment:
> The `--device_lead_allocator=` flag can be used in tooling to specify
a device that should be preferred as the allocator when a selection is
requested.

Might be worth mentioning this work in the upcoming release notes: https://github.com/iree-org/iree/issues/20722

Could link to one of the attached tracking issues and mention new APIs / flags.

Reply:
good idea - when the initial MVP of #20851 is working we can do that - there's no real way to use this now so it's a bit premature on this PR